src/ibl_to_nwb/tools/ecephys/spikeglxstreamerdatachunkiterator.py

```diff
@@ -0,0 +1,53 @@
+from typing import Optional, Tuple, Iterable
+
+from brainbox.io.spikeglx import Streamer
+from hdmf.data_utils import GenericDataChunkIterator
+
+
+class SpikeGLXStreamerDataChunkIterator(GenericDataChunkIterator):
```

Comment:
Before we over-commit to the streaming too early, like in MICrONS, can you do a quick speed check (in MB/s) to see what the downloads rate are between

(a) streaming via their `Streamer` class

and

(b) full file download via ONE (instructions were right below the streaming section as I recall). The latter would have the added benefit that we could just use the `SpikeGLXDataInterface`s instead of all that custom code

Reply:
Thanks - with decompression time then added to the download, I suppose they might be pretty close to each other.

Reply:
Hi.
Sorry I am reading this quickly.
We have in spiekinterface a extractor that decompress on the fly chunks.
https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/extractors/cbin_ibl.py
There is no need to decompress the entire file I think. Maybe I am missundertanding.

Reply:
Cool, thanks @samuelgarcia! This is very helpful, this way if we downloaded the compressed files from iBL we can just use SpikeGLXDataInterfaces. @CodyCBakerPhD what do you think?

Reply:
> There is no need to decompress the entire file I think.

I mean the time to decompress on the fly. If we're reading the entire file contents, that would add up.

Can do more testing but I'd be OK with the streaming approach here (which also means we can allocate like half as much space per AWS resource when it comes time for that).

Reply:
Oh I didn't realize we already have an extractor for this. Thanks for the pointer, @samuelgarcia. I think either approach could work. The advantage using that would be that we could make these interfaces more similar to our standard interface workflow.